File: system/frontend-sandbox/src/api/systemOperations.ts
import request from '@/utils/request'

export type ServicePurpose = 'text_review' | 'media_verification' | 'source_retrieval'

export interface ServiceConfig {
  id: number | null
  name: string
  purpose: ServicePurpose
  endpoint: string
  enabled: boolean
  credential_configured: boolean
  supports_media: boolean
  source: 'configured' | 'environment'
}

export interface OperationHealth {
  processing_count: number
  waiting_count: number
  completed_count: number
  attention_required_count: number
  message: string
}

export function getOperationHealth() {
  return request.get('/v2/system/operation-health')
}

export function listServiceConfigs() {
  return request.get('/v2/system/services')
}

export function createServiceConfig(body: {
  name: string
  purpose: ServicePurpose
  endpoint?: string
  service_identifier?: string
  protocol?: string
  credential?: string
  supports_media?: boolean
}) {
  return request.post('/v2/system/services', body)
}

export function setServiceEnabled(serviceId: number, enabled = true) {
  return request.post(`/v2/system/services/${serviceId}/activation`, { enabled })
}

export function checkServiceConnection(serviceId: number) {
  return request.post(`/v2/system/services/${serviceId}/connection-check`)
}

File: system/backend/app/api/v2/system_operations.py
"""Authenticated, product-safe system operation endpoints."""

from __future__ import annotations

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.security import require_roles
from app.db.mysql import get_db
from app.models.user import User
from app.schemas.system_operations import (
    ServiceActivationRequest,
    ServiceConfigCreateRequest,
)
from app.services import system_operations_service
from app.utils.response import success


router = APIRouter()


@router.get("/operation-health")
async def get_operation_health(
    _current_user: User = Depends(require_roles("admin", "analyst")),
    db: AsyncSession = Depends(get_db),
):
    return success(data=await system_operations_service.operation_health(db))


@router.get("/services")
async def list_services(
    _current_user: User = Depends(require_roles("admin", "analyst")),
    db: AsyncSession = Depends(get_db),
):
    return success(data=await system_operations_service.list_service_configs(db))


@router.post("/services")
async def create_service(
    body: ServiceConfigCreateRequest,
    current_user: User = Depends(require_roles("admin")),
    db: AsyncSession = Depends(get_db),
):
    try:
        result = await system_operations_service.create_service_config(
            payload=body.model_dump(),
            user_id=int(current_user.id),
            db=db,
        )
        await db.commit()
    except ValueError as exc:
        raise HTTPException(status_code=400, detail=str(exc)) from exc
    return success(data=result)


@router.post("/services/{service_id}/activation")
async def activate_service(
    service_id: int,
    body: ServiceActivationRequest,
    _current_user: User = Depends(require_roles("admin")),
    db: AsyncSession = Depends(get_db),
):
    try:
        result = await system_operations_service.activate_service_config(
            service_id=service_id,
            enabled=body.enabled,
            db=db,
        )
        await db.commit()
    except ValueError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc
    return success(data=result)


@router.post("/services/{service_id}/connection-check")
async def check_service_connection(
    service_id: int,
    _current_user: User = Depends(require_roles("admin")),
    db: AsyncSession = Depends(get_db),
):
    try:
        result = await system_operations_service.check_service_config(service_id, db)
    except ValueError as exc:
        raise HTTPException(status_code=404, detail=str(exc)) from exc
    return success(data=result)


__all__ = ["router"]
